scrapy/commands/genspider.py

```diff
@@ -69,7 +69,23 @@ def run(self, args, opts):
         try:
```

Comment:
I’m thinking you could extract the `opts.force` check out of the whole `try`-`except`, since the whole `try`-`except` is only there to check if the spider or file already exist:

```python
if not opts.force:
    try:
    #…
```

Also, since this part of the code is getting long, maybe it would make sense to take the check into its own method:

```python

if not opts.force and self._spider_exists(name):
    return

# …

def _spider_exists(self, name):
    # …
```

With this later approach, you could leave `pass` within `except KeyError`, and move the additional checks _after_ the `else` of the `try`-`except`, using early returns to minimize indentation.

Reply:
Did you mean something like this?
```py
def _spider_exists(self, name):
        if not self.settings.get('NEWSPIDER_MODULE'):
            # if run as a standalone command and file with same filename already exists
            if exists(name + ".py"):
                print("%s already exists" % (abspath(name + ".py")))
                return True
        else:
            try:
                spidercls = self.crawler_process.spider_loader.load(name)
            except KeyError:
                pass
            else:
                # if spider with same name exists
                print("Spider %r already exists in module:" % name)
                print("  %s" % spidercls.__module__)
                return False
            # if spider same filename exists
            spiders_module = import_module(self.settings['NEWSPIDER_MODULE'])
            spiders_dir = dirname(spiders_module.__file__)
            spiders_dir_abs = abspath(spiders_dir)
            if exists(join(spiders_dir_abs, name + ".py")):
                print("%s already exists" % (join(spiders_dir_abs, (name + ".py"))))
                return True

        return False
```
Then the entire block including (L69-94) can be reduced to 
```py
if not opts.force and self._spider_exists(name): 
    return
```